### DISCUSSION TITLE
JSPI-enabled emscripten_sleep() gets "invalid suspender object for suspend" error

### DISCUSSION THREAD

**Discussion by rentallect** [emscripten-core/emscripten#20407] [Q&A] [answered]
I am using emscripten `3.1.46`.  I build my wasm with `-s ASYNCIFY=2` enabled.  In the C code that is transformed into my wasm, I have a call to `emscripten_sleep(5)`.  

At runtime, when the `emscripten_sleep` function is invoked, it triggers a `RuntimeError: invalid suspender object for suspend`.

Can someone give me some clues regarding what I might be neglecting to do to make this `emscripten_sleep` operate as expected?

**Comment by brendandahl** [comment, marked as answer]
This is likely because the wasm export that eventually lead to the call of sleep did not have JSPI applied. Can you see on the call stack where wasm was entered from JS? That function should be `ASYNCIFY_EXPORTS` in your build command.
